### PR TITLE
Add #4696 (dynamic backend, dynamic flatten) after revert

### PR DESCRIPTION
**Pull request by mbencer**
Attempt of merge https://github.com/NervanaSystems/ngraph/pull/4696 again

### CODE REVIEW

**Comment by mbencer** [comment]
OV tests: https://gitlab-icv.inn.intel.com/inference-engine/dldt/merge_requests/7468

File: python/src/ngraph/impl/__init__.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
 from _pyngraph import Type
 from _pyngraph import PartialShape
 from _pyngraph import Shape
+from _pyngraph import PartialShape
```
Comment (silee2) [inline python/src/ngraph/impl/__init__.py:41]: Duplicate import(line 39).

**Comment by mbencer** [comment]
wait for https://github.com/NervanaSystems/ngraph-onnx/pull/621 merge

File: python/src/ngraph/impl/__init__.py (hunk 1 above)
Comment (mbencer) [reply python/src/ngraph/impl/__init__.py]: redundant line removed

**Pull request merged**